File: RepoTransBench/target_projects/Python/Java/thomasa88_ParametricText/src/test/java/com/thomasa88/original/TestParamFormatter.java
package com.thomasa88.original;

import com.thomasa88.DummyParam;
import com.thomasa88.DummyDesign;
import com.thomasa88.ParamFormatter;
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class TestParamFormatter {

    private DummyDesign design;

    @BeforeEach
    void setUp() {
        design = new DummyDesign();
    }

    @Test
    void testUnitlessPositive() {
        DummyParam p = new DummyParam(1.75, "");
        assertEquals("1 3/4\"", ParamFormatter.mixedFracInch(p, design));
    }

    @Test
    void testUnitlessNegative() {
        DummyParam p = new DummyParam(-2.5, "");
        assertEquals("-2 1/2\"", ParamFormatter.mixedFracInch(p, design));
    }

    @Test
    void testUnitInch() {
        DummyParam p = new DummyParam(2.5, "in");
        assertEquals("2 1/2\"", ParamFormatter.mixedFracInch(p, design));
    }

    @Test
    void testWholeNumber() {
        DummyParam p = new DummyParam(3.0, "");
        assertEquals("3\"", ParamFormatter.mixedFracInch(p, design));
        DummyParam p2 = new DummyParam(0.0, "");
        assertEquals("0\"", ParamFormatter.mixedFracInch(p2, design));
    }

    @Test
    void testFractionOnly() {
        DummyParam p = new DummyParam(0.25, "");
        assertEquals("1/4\"", ParamFormatter.mixedFracInch(p, design));
        DummyParam p2 = new DummyParam(-0.75, "");
        assertEquals("-3/4\"", ParamFormatter.mixedFracInch(p2, design));
    }

    @Test
    void testZero() {
        DummyParam p = new DummyParam(0, "");
        assertEquals("0\"", ParamFormatter.mixedFracInch(p, design));
    }
}

File: RepoTransBench/source_projects/Python/thomasa88_ParametricText/test/test_paramformatter.py
import unittest
from paramformatter import mixed_frac_inch

class DummyUnitsManager:
    def convert(self, value, from_unit, to_unit):
        # For test, just return the input value unchanged.
        return value

class DummyParam:
    def __init__(self, value, unit):
        self.value = value
        self.unit = unit

class DummyDesign:
    def __init__(self):
        self.fusionUnitsManager = DummyUnitsManager()

class TestMixedFracInch(unittest.TestCase):
    def setUp(self):
        self.design = DummyDesign()

    def test_unitless_positive(self):
        p = DummyParam(1.75, '')
        self.assertEqual(mixed_frac_inch(p, self.design), '1 3/4"')

    def test_unitless_negative(self):
        p = DummyParam(-2.5, '')
        self.assertEqual(mixed_frac_inch(p, self.design), '-2 1/2"')

    def test_unit_inch(self):
        p = DummyParam(2.5, 'in')
        # Should use units manager, but here conversion returns value unchanged
        self.assertEqual(mixed_frac_inch(p, self.design), '2 1/2"')

    def test_whole_number(self):
        p = DummyParam(3.0, '')
        self.assertEqual(mixed_frac_inch(p, self.design), '3"')
        p = DummyParam(0.0, '')
        self.assertEqual(mixed_frac_inch(p, self.design), '0"')

    def test_fraction_only(self):
        p = DummyParam(0.25, '')
        self.assertEqual(mixed_frac_inch(p, self.design), '1/4"')
        p = DummyParam(-0.75, '')
        self.assertEqual(mixed_frac_inch(p, self.design), '-3/4"')

    def test_zero(self):
        p = DummyParam(0, '')
        self.assertEqual(mixed_frac_inch(p, self.design), '0"')